yarn-project/aztec-nr/aztec/src/oracle/get_low_nullifier_membership_witness.nr

```diff
@@ -0,0 +1,46 @@
+use crate::constants_gen::NULLIFIER_TREE_HEIGHT;
+use crate::utils::arr_copy_slice;
+use crate::hash::pedersen_hash;
```

Comment:
this and get_membership witness could be combined with generics and traits

Reply:
Given that NullifierMembershipWitness contains leaf_data I am not sure this makes sense. WDYT?
<img width="813" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/13470840/44b23451-7128-4141-a8ef-e32da3106b0c">

Reply:
You could to nullifierMemberShipWitness that includes a membership witness since both the index and path are shared 🤷

Reply:
@LHerskind that would not allow you to have one get_membership_witness oracle so I don't see that much value in doing that.

Reply:
leaf_data is just a pre_image, so combining the two just means adjusting the interface to display that?